framework/moose.mk

```diff
@@ -315,7 +315,7 @@ $(pcre_LIB): $(pcre_objects)
 $(gtest_LIB): $(gtest_objects)
 	@echo "Linking Library "$@"..."
 	@$(libmesh_LIBTOOL) --tag=CC $(LIBTOOLFLAGS) --mode=link --quiet \
-	  $(libmesh_CC) $(libmesh_CFLAGS) -o $@ $(gtest_objects) $(libmesh_LDFLAGS) $(libmesh_LIBS) $(EXTERNAL_FLAGS) -rpath $(gtest_DIR)
+	  $(libmesh_CC) -o $@ $(gtest_objects) $(EXTERNAL_FLAGS) -rpath $(gtest_DIR)
```

Comment:
The immediate reason is that when doing code coverage we add `-fcode-coverage` to `libmesh_CFLAGS` so if you have `libmesh_CFLAGS` then you must also have `libmesh_LDFLAGS` which will throw in a ton of unnecessary library paths pointing to libmesh, petsc, etc. libraries that gtest should care nothing about. I changed this entire compile/link line because an individual gtest library would link to whatever the first libmesh METHOD library you compiled. And then if you compiled MOOSE unit with a different METHOD you would actually get a binary linked to two different libmesh library versions, leading to segmentation faults on program exit.

If someone wants custom CFLAGS for compiling `gtest` then I think the solution is to add a `gtest_CFLAGS` and not to keep using flags that are meant for libmesh derived libraries/applications.